pkg/controller/staticroute/staticroute_controller.go

```diff
@@ -89,7 +88,55 @@ func add(mgr manager.Manager, r reconcile.Reconciler) error {
 	}
 
 	// Watch for changes to primary resource StaticRoute
-	return c.Watch(&source.Kind{Type: &iksv1.StaticRoute{}}, &handler.EnqueueRequestForObject{})
+	err = c.Watch(&source.Kind{Type: &iksv1.StaticRoute{}}, &handler.EnqueueRequestForObject{})
+	if err != nil {
+		return err
+	}
+
+	// Watch if the self node labels are changed, so reconcile every route
+	err = c.Watch(
+		&source.Kind{Type: &corev1.Node{ObjectMeta: metav1.ObjectMeta{Name: r.(*ReconcileStaticRoute).options.Hostname}}},
+		&handler.EnqueueRequestsFromMapFunc{
+			ToRequests: handler.ToRequestsFunc(func(a handler.MapObject) []reconcile.Request {
+				routes := &iksv1.StaticRouteList{}
+				if err := r.(*ReconcileStaticRoute).client.List(context.Background(), routes); err != nil {
+					//TODO escalate the problem somehow
```

Comment:
Some logging would be nice here

Reply:
I know, but this is a special place, where we do not have any requestlogger, cannot return any error, etc..
Should I log with some default `log` package if the `List` call fails?

Reply:
Put some logs. Also to the places where label change is detected.